packages/core/src/domain/internalMonitoring.ts

```diff
@@ -32,7 +32,7 @@ const monitoringConfiguration: {
   sentMessageCount: number
 } = { maxMessagesPerPage: 0, sentMessageCount: 0 }
 
-let externalContextProvider: () => Context
+export let externalContextProvider: () => Context
```

Comment:
Mutable export is fun but a bit confusing :) I think you could either:
* Consolidate the Message with the Context in this module instead of in the batch, so `externalContextProvider` stays local to this module
* Pass the `externalContextProvider` to the batch by using an indirection, ex: `startBatch(() => externalContextProvider?.())`

Reply:
Right, and in fact I was also missing the external context when sending to the event bridge. 
It's now fixed. 
Does it look good to you?

Reply:
Looks good!